src/node.cc

```diff
@@ -1903,7 +1944,7 @@ static void InitGroups(const FunctionCallbackInfo<Value>& args) {
 
 
 void Exit(const FunctionCallbackInfo<Value>& args) {
-  exit(args[0]->Int32Value());
+  Environment::GetCurrent(args)->Exit(args[0]->Int32Value());
```

Comment:
I think it would help code clarity to keep the exit() call for the main isolate here.